duckdb-vortex/src/vortex_scan.cpp

```diff
@@ -249,20 +264,28 @@ static void CreateScanPartitions(ClientContext &context, const BindData &bind, S
 	const auto partition_count = std::max(static_cast<uint64_t>(1), row_count / partition_size);
 	global_state.partitons_total += partition_count;
 
-	// Pin threads to files as long as there more files than threads left to process.
-	const bool pin_file_to_thread = (file_count - global_state.files_partitioned) > thread_count;
-	auto &partitions_queue = pin_file_to_thread ? local_state.scan_partitions : global_state.scan_partitions;
-
-	for (size_t partition_idx = 0; partition_idx < partition_count; ++partition_idx) {
+	if (PinFileToThread(global_state)) {
+		local_state.thread_local_file_idx = file_idx;
 
-		partitions_queue.enqueue(ScanPartition {
-		    .file_idx = file_idx,
-		    .start_row = partition_idx * partition_size,
-		    .end_row = (partition_idx + 1) == partition_count ? row_count : (partition_idx + 1) * partition_size,
-		});
+		for (size_t partition_idx = 0; partition_idx < partition_count; ++partition_idx) {
+			local_state.scan_partitions.push(ScanPartition {
+			    .file_idx = file_idx,
+			    .start_row = partition_idx * partition_size,
+			    .end_row = (partition_idx + 1) == partition_count ? row_count : (partition_idx + 1) * partition_size,
+			});
+		}
+	} else {
+		for (size_t partition_idx = 0; partition_idx < partition_count; ++partition_idx) {
+			global_state.scan_partitions.enqueue(ScanPartition {
+			    .file_idx = file_idx,
+			    .start_row = partition_idx * partition_size,
+			    .end_row = (partition_idx + 1) == partition_count ? row_count : (partition_idx + 1) * partition_size,
+			});
```

Comment:
This would read must easier to read if you swap the if and the for

Reply:
done